src/Chapter1.hs

```diff
@@ -653,7 +668,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit n
+        | abs n < 10 = abs n
+        | abs n < 0 = firstDigit(abs n)
+        | otherwise = firstDigit(abs n `div` 10)
```

Comment:
That is right 💯 

You can see that `abs n` is reused several times in this function. This usually means that you can move it into the `where` case and reuse everywhere 👍🏼 

Also note, that `abs n` will never be lower than 0, so this check is redundant.

And in Haskell functions and operators are always separated with spaces from the arguments, even if arguments are in parenthesis 👾 

```haskell
        | abs n < 10 = abs n
        | otherwise = firstDigit (abs n `div` 10)
```